DietPi-Software | Docker

techdabbler opened the pull request:
Added Docker installation/config/removal.

Things to note -- 

ARM Only - I haven't tested it on x86_64 yet so it is currently disabled.  No reason to believe it will not work just haven't thrown up a VM yet to test it in.

Index number is based assuming the fuguhub install will be merged. So if this one goes in before fuguhub then the index will need to be reduced as well as the hard limit.  Not sure if this is the best way to handle it so let me know if there is a better method.

In partial response to issue https://github.com/Fourdee/DietPi/issues/870. Not sure how to link these correctly.

Commit:
added docker

index is assumed fuguhub is added so merge after that

not tested on x86_64 yet so it isn't enabled

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -7012,6 +7021,29 @@ _EOF_
+
+			if (( $? == 0 )); then
+				# Offical Docker recommended install command
+				wget -O DockerInstall.sh https://get.docker.com
```
Minor:
You can use the variable
```wget -O DockerInstall.sh "$INSTALL_URL_ADDRESS"```

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -11590,6 +11622,30 @@ _EOF_
+			sed -i '/ExecStart=\/usr\/bin\/dockerd -H fd:\/\// c\ExecStart=\/usr\/bin\/dockerd -g \/mnt\/dietpi_userdata\/docker-data -H fd:\/\/' /lib/systemd/system/docker.service
+
+			# Add to service for dietpi
+			cat << _EOF_ >> /DietPi/dietpi/.dietpi-services_include
```
I'd prefer if we hard code ```'docker'``` to ```dietpi-services```:
https://github.com/Fourdee/DietPi/blob/master/dietpi/dietpi-services#L105

We do this for all Software services available in ```dietpi-software```.

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -12604,6 +12660,19 @@ _EOF_
 
+		elif (( $1 == 162)); then
+			# remove docker and all its unused dependencies
+			apt-get remove --auto-remove -y docker-engine
```
You can use function ```AGP docker-engine``` here, it will purge.
```dietpi-software``` also runs ```apt-get autoremove --purge``` automatically, at end of uninstalls

Commit:
Updated fourdee recs

Still need to test he url variable I think it didn't work which is why i hard coded it. Still to test.

Commit:
Merge pull request #4 from techdabbler/fuguhub-patch

Merge Fuguhub patch

Commit:
Merge branch 'dietpisoftware-all-testing' into docker-add

ghost commented:
I messed up my branch  I merged branches the wrong way. So hang tight until I fix this.

Commit:
reversion attempt

mostly removes fuguhub which is what concerned me in the first place.

ghost commented:
Ok my branch is back to where it was before. Sorry about that merge mess.

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -12604,6 +12655,15 @@ _EOF_
+			AGP docker-engine
+
+			# delete data files - dietpi
+			rm -r FP_DIETPI_USERDATA_DIRECTORY/docker-data
```
Missing the variable ```$```, should be:
```
rm -r "$FP_DIETPI_USERDATA_DIRECTORY"/docker-data
```

Fourdee commented on dietpi/dietpi-software:
```diff
@@ -11590,6 +11622,25 @@ _EOF_
+			#usermod -aG docker docker-user
+
+			# Create directory for docker containers
+			mkdir /mnt/dietpi_userdata/docker-data
```
Please replace ```/mnt/dietpi_userdata``` with the variable:
```
"$FP_DIETPI_USERDATA_DIRECTORY"
```

Fourdee reviewed (commented):
Few minor changes outstanding, before we can merge. Please see review below.

Fourdee commented:
Note to self:
Once changes completed. Merge https://github.com/Fourdee/DietPi/pull/1168 1st.

Fourdee reviewed (changes requested):
Changes required:
https://github.com/Fourdee/DietPi/pull/1169#pullrequestreview-67835509

Commit:
syntax cleanup

ghost commented:
@Fourdee Done

Fourdee committed:
Merge branch 'testing' into docker-add

Fourdee commented:
@techdabbler 
Legend, many thanks 👍
